src/components/Board.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import Square from './Square';
+import BoardContext from '../context/BoardContext'
+
+function Board(props) {
+    const { board, setBoard, turn, setTurn, history, setHistory } = useContext(BoardContext);
+
+    function declareWinner(board) {
+        const winningCombos = [
+            [3, 4, 5],
+            [0, 1, 2],
+            [0, 3, 6],
+            [2, 5, 8],
+            [1, 4, 7],
+            [0, 4, 8],
+            [2, 4, 6],
+            [6, 7, 8],
+        ];
+
+        function threeInARow() {
+            for (let i = 0; i < winningCombos.length; i++) {
+                const [first, second, third] = winningCombos[i];
+                // When you get 3 in a row, in any direction.
+                if (board[first] && board[first] === board[second] && board[first] === board[third]) {
+                    return board[first];
+                }
+            }
+            return null;
+        }
+        threeInARow()
+    };
+
+    const winner = declareWinner(board);
+
+    // Check for winner, every move.
+    let status = null;
+    if (winner) {
+        status = `${winner} WON THE GAME!`;
+    };
+
+    function timeTravel(turn) {
+            const boardInHistory = [...history];
+            setBoard(boardInHistory[turn]);
+    }
+    
+    function clickHandler(i) {
```

Comment:
No right or wrong answer here, but I think you could make an argument that this logic could fit nicely on the context. Instead of the context providing `setBoard`, `setHistory`, and `setTurn`, you could just pass this function and let it get called as the callback. The downside if your context becomes less flexible, but the upside is that the component stays shorter.

Reply:
One other upside to that approach is that the `Square` component could pull it's callback directly from the context instead of requiring it as a prop.